source/docs/hardware/hardware-safety/io_safety.rst

```diff
@@ -0,0 +1,27 @@
+Input/Output Safety Mechanisms Built into the `FRC` Control System
+============================================================================
+
+roboRIO Control System
+^^^^^^^^^^^^^^^^^^^^^^
+
+There are multiple safety mechanisms on the robot that handle input / output operations while it is powered on.
+
+Robot side: there are multiple hardware and software components involved. Outputs of the RoboRIO (e.g. :term:`PWM` s) are controlled by the :term:`FPGA` hardware. :term:`NetComm` is a software daemon that talks to the DS, the FPGA, and the user program. Inside of the user process \(the team\’s robot program\), there\’s a NetComm :term:`DLL` component that talks to the FPGA, CAN, and the NetComm daemon. And of course there are CAN motor controllers on the CAN bus.
+
+- The FPGA has a system :term:`watchdog`. This watchdog will time out and force a “disable” of PWM motor outputs if NetComm hasn\’t told it it\’s received an enable packet in the last 125 `ms`.
+- The NetComm DLL in the user process will send a disable broadcast message on the CAN network and then stop sending keep-alive CAN messages after the disabled system watchdog signal is read back from the FPGA \(this is checked on a 20 `ms` loop\). REV pneumatics and motor controllers will stop immediately upon receipt of the disable broadcast. They also stop if no keep-alive is received for 100 `ms` \(pneumatics\) or 220 `ms` \(motor controllers\).
```

Comment:
Additionally, there is incorrect information about how the roboRIO's software behaves in this paragraph. While the roboRIO is booted, it never stops sending [Universal Heartbeat](https://docs.wpilib.org/en/stable/docs/software/can-devices/can-addressing.html#universal-heartbeat) messages. When the robot is stopped, the roboRIO merely changes the `WatchdogEnabled` field from a 1 to a 0, but it keeps sending out the heartbeat. 

This paragraph should also probably use the official "heartbeat" term consistently, rather than "keep-alive", which I think more strongly implies that receipt of the frame itself indicates an enabled status, rather than the truth, which is that it merely indicates the presence of a roboRIO, and you have to look at a specific byte to check the enabled status.

Reply:
Our devices will also stop immediately when they receive a heartbeat with the `WatchdogEnabled` field set to 0, of course.

Reply:
Not to block the PR, but I think it would be _nice_ to have the timeout value(s) documented. Even if implementations will vary and even change, would it be fair to document minimum and maximum timeouts that all implementations fall inside of?

Reply:
I think its important to have those timeouts (or at least an upper bound) documented. @Kevin-OConnor maybe you can comment on what FIRST's expectation of motor controllers timeout period is?

Reply:
It seems odd that this specifically calls out what REVs devices do while not saying anything about other vendors. This should probably be genericized and use language similar to the "brief timeout period" Noah has suggested.